plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1787,6 +1810,28 @@ private void checkUnsupportedGeneratedColumns(MetadataEntry metadataEntry)
         }
     }
 
+    private void checkUnsupportedColumnMappingIdStructuralTypes(MetadataEntry metadataEntry)
+    {
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry);
+        boolean hasTableStructuralTypes = getColumns(metadataEntry).stream()
+                .filter(column -> HiveUtil.isStructuralType(column.getPhysicalType())).findFirst().isPresent();
```

Comment:
No need to call this line when the column mapping isn't id.

Reply:
fixed